### PR TITLE
Adding in support for kits, refactor healing, and BIG rework on damage 

### PR DESCRIPTION
**Pull request by nhillen**
Only strike uses the new damage code currently, we need to refactor the other things that do damage to do something similar (and we'll need to add something for cleave to ignore armor)

### CODE REVIEW

File: eldritchmush/commands/combatant.py [hunk 1]
```diff
@@ -274,3 +278,98 @@ def hasChirurgeonsKit(self):
...
+                else:
+                    new_value = self.caller.attributes.get(type) - amount
+
+                obj_to_set = self.caller.attributes.get(type, return_obj=True)
+                obj_to_set.value = new_value
+
+        return remaining_damage
+
+    def updateAv(self):
+        self.caller.db.av = self.caller.db.shield_value + self.caller.db.armor + self.caller.db.tough + self.caller.db.armor_specialist
```
Comment (spencermcghin) [inline eldritchmush/commands/combatant.py:387]: You need to add indomitable to this calc and anywhere you update av

**Review by spencermcghin — COMMENTED** [review]
Add one comment

File: eldritchmush/commands/combatant.py [hunk 2]
```diff
@@ -274,3 +278,98 @@ def hasChirurgeonsKit(self):
...
+                return self.alternateDamage(amount, "death_points")
+            else:
+                return 0
+        else:
+            return self.alternateDamage(amount, "death_points")
+
+    def alternateDamage(self, amount, type):
+        remaining_damage = amount
+
+        if self.caller.attributes.get(type):
```
Comment (spencermcghin) [inline eldritchmush/commands/combatant.py:368]: Did you mean to do two ifs here?

File: eldritchmush/commands/combatant.py (hunk 1 above)
Comment (nhillen) [reply eldritchmush/commands/combatant.py:387]: Cool, I was just grabbing from Code so it’s probably missing in a few places, should indomitable get decreased like tough when you take damage?

File: eldritchmush/commands/combatant.py (hunk 2 above)
Comment (nhillen) [reply eldritchmush/commands/combatant.py:368]: I think so!   What looks weird?

**Pull request merged**